e2e/pod.go

```diff
@@ -59,6 +59,11 @@ func waitForDeploymentComplete(name, ns string, c kubernetes.Interface, t int) e
 	err = wait.PollImmediate(poll, timeout, func() (bool, error) {
 		deployment, err = c.AppsV1().Deployments(ns).Get(context.TODO(), name, metav1.GetOptions{})
 		if err != nil {
+			if errors.Is(err, wait.ErrWaitTimeout) {
```

Comment:
can you try with
```go
if testutils.IsRetryableAPIError(err) {
				return false, nil
}
```

This check is already present in `waitForDaemonSets` can we include this in other places also where ever we are pooling for the response

Reply:
sure!